File: suites/osprey-compute-kernels/reference/cases/hanoi/hanoi.c
#include <stdint.h>
#include <stdio.h>

int64_t hanoi(int64_t n, int64_t acc) {
    if (n == 0) {
        return acc;
    }
    return hanoi(n - 1, hanoi(n - 1, acc) + 1);
}

int main(void) {
    printf("%lld\n", (long long)hanoi(25, 0));
    return 0;
}

File: suites/osprey-compute-kernels/reference/cases/hanoi/hanoi.rs
fn hanoi(n: i64, acc: i64) -> i64 {
    if n == 0 {
        acc
    } else {
        hanoi(n - 1, hanoi(n - 1, acc) + 1)
    }
}

fn main() {
    println!("{}", hanoi(25, 0));
}
